torch/_inductor/decomposition.py

```diff
@@ -357,10 +358,29 @@ def all_dim(input, dim, keepdim=False):
 @register_decomposition(aten.copy)
 def copy(self, src, non_blocking=False):
     intermediate = src.to(self, non_blocking)
-    if self.size() != intermediate.size():
-        return aten.expand_copy.default(intermediate, self.size())
-    else:
+    # cheapest case
+    if self.size() == intermediate.size() and self.stride() == intermediate.stride():
```

Comment:
You shoulds check storage offset as well here I think?
Also in all the conditions below as contiguous =/> storage_offset == 0